src/common/http_defs.cc

```diff
@@ -62,9 +62,10 @@ CacheDirective::delta() const
             return std::chrono::seconds(data.maxStale);
         case MinFresh:
             return std::chrono::seconds(data.minFresh);
+        default:
+            throw std::domain_error("Invalid operation on cache directive");
+            return std::chrono::seconds();
```

Comment:
This return will never execute.  Move it outside of the switch statement?  It won't execute their either...  I'm surprised that the compiler is not throwing a warning about missing a return statement.  Must be smart enough to realize that all cases are covered and that all will exit the method before the normal exit at the bottom.

I'm ambivalent on this one.  It just looks weird I guess.

Reply:
You are right, it looks weird. I just tried removing the return statement and I no warning appeared. I'll remove the return statement.